### PR TITLE
Add Error Message in RegistrationPage and Show User Info in SettingsScreen

### PR DESCRIPTION
**Pull request by KevinZxy1273**
When user registration information doesn't pass the valid check by backend, corresponding error message will be shown.

Once the user is logged in the username and avatar will be shown in the SettingsScreen.

### CODE REVIEW

**Comment by greatfighter** [comment]
Looks good

**Review by greatfighter — APPROVED** [review]

**Review by marklhj-lee — APPROVED** [review]
LGTM

File: app/src/main/java/com/hydrofish/app/ui/composables/tabs/SettingsScreen.kt [hunk 1]
```diff
@@ -119,6 +130,49 @@ fun SettingsScreen(permissionChecker: PermissionChecker,
...
                 .padding(paddingValues)
         ) {
+
+            if (isUserLoggedIn && userName != null) {
+                Box(
+                    contentAlignment = Alignment.Center,
+                    modifier = Modifier.fillMaxWidth()
+                ) {
+//                    Image(
+//                        painter = rememberAsyncImagePainter(model = R.drawable.user_avatar), // 替换为实际的头像URL或资源ID
```
Comment (isabellachen3988) [inline app/src/main/java/com/hydrofish/app/ui/composables/tabs/SettingsScreen.kt:140]: did you mean to keep this in?

**Review by isabellachen3988 — APPROVED** [review]
seems good, please have a look at my comment though

**Pull request merged**